lessons/creating-apis-with-python-and-flask.md

```diff
@@ -0,0 +1,713 @@
+---
+title: |
+    Creating Web APIs with Python and Flask
+collection: lessons
+layout: lesson
+slug: creating-apis-with-python-and-flask
+date: 2018-04-02
+authors:
+- Patrick Smyth
+reviewers:
+- Megan Kudzia
+- Ryan Deschamps
+editors:
+- Brandon Walsh
+difficulty: 2
+activity: presenting
+topics: [api, data-management]
+review-ticket:  106
+abstract: |
+  Learn how to set up a basic Application Programming Interface (API) to make your data more accessible to users. This lesson also discusses principles of API design and the benefits of APIs for digital projects.
+---
+
+{% include toc.html %}
+
+# Lesson Goals
+
+Web APIs are tools for making information and application functionality accessible over the internet. In this lesson, you will:
+
+- Learn what an API is and when you should use one.
+- Learn how to build a web API that returns data to its users.
+- Learn some principles of good API design, applying them to an API that draws book metadata from a database.
+
+# Setting Up
+
+## Prerequisites
+
+You can use the Windows, OSX, or Linux operating systems to complete this tutorial, and those few instructions that are not the same across platforms will be explicitly noted. Python 3, the Flask web framework, and a web browser are required for this tutorial, and installation instructions for all platforms are outlined below.
+
+The only knowledge explicitly assumed for this lesson is the ability to use a text editor, such as BBEdit on OSX or Notepad++ on Windows. However, knowledge of the command line, Python, and web concepts such as HTTP may make this tutorial easier to follow. If you're new to Python, consider working through the Programming Historian series on [dealing with online sources](https://programminghistorian.org/lessons/introduction-and-installation) to familiarize yourself with fundamental concepts in Python programming.
+
+## Installing Python and Flask
+
+For this tutorial, you will need Python 3 and the Flask web framework. You'll also require a web browser (such as Firefox) and a text editor (such as Notepad++ or BBEdit).
+
+To download Python, follow [this link](https://www.python.org/downloads/), select the button that says `Download Python 3.x.x`, and then run the installer as you normally would to install applications on your operating system. The default settings should be fine.
+
+To confirm that Python installed successfully, first open the command line. In OSX, click the finder on the top left of your desktop (the magnifying glass) and type `terminal`. The terminal should be the first application that appears. On Windows, click the Start menu icon and type `cmd` in the search box, then press `Enter`.
```

Comment:
Pedantic: "finder on the top left of your desktop (the magnifying glass)" --> this is "spotlight," right? Finder is the file browser, and spotlight is in the upper-right corner. 😄 

Maybe change to

"In macOS, click the spotlight icon on the top right corner of your desktop (the magnifying glass) and type `terminal`."